src/backend/backend_common.c

```diff
@@ -362,6 +362,100 @@ struct conv **generate_blur_kernel(enum blur_method method, void *args, int *ker
+	    {.iterations = 1, .offset = 2.25f},        // LVL  2 => radius   7
+	    {.iterations = 2, .offset = 2.00f},        // LVL  3 => radius  14
+	    {.iterations = 2, .offset = 3.00f},        // LVL  4 => radius  20
+	    {.iterations = 2, .offset = 4.25f},        // LVL  5 => radius  28
+	    {.iterations = 3, .offset = 2.50f},        // LVL  6 => radius  35
+	    {.iterations = 3, .offset = 3.25f},        // LVL  7 => radius  45
+	    {.iterations = 3, .offset = 4.25f},        // LVL  8 => radius  57
+	    {.iterations = 3, .offset = 5.50f},        // LVL  9 => radius  74
+	    {.iterations = 4, .offset = 3.25f},        // LVL 10 => radius  91
+	    {.iterations = 4, .offset = 4.00f},        // LVL 11 => radius 110
+	    {.iterations = 4, .offset = 5.00f},        // LVL 12 => radius 135
+	    {.iterations = 4, .offset = 6.00f},        // LVL 13 => radius 161
+	    {.iterations = 4, .offset = 7.25f},        // LVL 14 => radius 195
+	    {.iterations = 4, .offset = 8.25f},        // LVL 15 => radius 221
+	    {.iterations = 5, .offset = 4.50f},        // LVL 16 => radius 250
+	    {.iterations = 5, .offset = 5.25f},        // LVL 17 => radius 287
+	    {.iterations = 5, .offset = 6.25f},        // LVL 18 => radius 330
+	    {.iterations = 5, .offset = 7.25f},        // LVL 19 => radius 383
+	    {.iterations = 5, .offset = 8.50f},        // LVL 20 => radius >450
+	};
+
+	auto params = ccalloc(1, struct dual_kawase_params);
+	params->iterations = 0;
+	params->offset = 1.0f;
+
+	if (blur_args->strength <= 0 && blur_args->size) {
+		// approximate blur_strength with gaussian blur_radius
+		if (blur_args->size < 6) {
+			blur_args->strength = 1;
+		} else if (blur_args->size < 11) {
+			blur_args->strength = 2;
+		} else if (blur_args->size < 17) {
+			blur_args->strength = 3;
+		} else if (blur_args->size < 24) {
+			blur_args->strength = 4;
+		} else if (blur_args->size < 32) {
+			blur_args->strength = 5;
+		} else if (blur_args->size < 40) {
+			blur_args->strength = 6;
+		} else if (blur_args->size < 51) {
+			blur_args->strength = 7;
+		} else if (blur_args->size < 67) {
+			blur_args->strength = 8;
+		} else if (blur_args->size < 83) {
+			blur_args->strength = 9;
+		} else if (blur_args->size < 101) {
+			blur_args->strength = 10;
+		} else if (blur_args->size < 123) {
+			blur_args->strength = 11;
+		} else if (blur_args->size < 148) {
+			blur_args->strength = 12;
+		} else if (blur_args->size < 177) {
+			blur_args->strength = 13;
+		} else if (blur_args->size < 208) {
+			blur_args->strength = 14;
+		} else if (blur_args->size < 236) {
+			blur_args->strength = 15;
+		} else if (blur_args->size < 269) {
+			blur_args->strength = 16;
+		} else if (blur_args->size < 309) {
+			blur_args->strength = 17;
+		} else if (blur_args->size < 357) {
+			blur_args->strength = 18;
+		} else if (blur_args->size < 417) {
+			blur_args->strength = 19;
+		} else {
+			blur_args->strength = 20;
+		}
```

Comment:
This could be possible with a lookup-table and a for-loop as well. Not sure what might be cleaner.

Reply:
I think a lookup-table would be better.